### ISSUE TITLE
Feature Request: Support "Link with phone number" (Pairing Code) for TTY/Accessibility

### ISSUE THREAD

**Issue by jessy836** [xenodium/wasabi#10]
hello,

I use Emacs in a terminal (TTY) environment. Currently, Wasabi requires scanning a QR code to log in. This presents a blocking issue for me because:
 1. I am visually impaired and use a screen reader (NVDA), so I cannot see the QR code.
 2. In a TTY environment, the QR image cannot be rendered or scanned properly.

Would it be possible to add the "Link with phone number" option (where the client generates an 8-character pairing code)? This would allow authentication without a camera/QR code.

Thanks!

**Comment by xenodium** [comment]
Hi @jessy836 thanks for filing. Let see what I can do for this. Depends on what [wuzapi](https://github.com/asternic/wuzapi) provides which wasabi relies on. On the surface, it looks like it's possible, but I need to prototype.

Heads-up, wasabi still has quite a few rough edges (I'm guessing possibly more on TTY since I'm using GUI Emacs).

**Comment by xenodium** [comment]
I've just pushed https://github.com/xenodium/wasabi/commit/cf58abf0ed853fafe100f7bd186a71de75db9f86 commit which adds an interactive command `wasabi-pair-phone-number command`.

When the QR code is rendered in the buffer, it should be accompanied by text or message like:

"Scan from WhatsApp mobile to enable client or invoke M-x wasabi-pair-phone-number"

Only when that message is in the buffer, invoking `M-x wasabi-pair-phone-number` will work. This function asks for a phone number in the form +1234567890 and return the code into the minibuffer. The code is also saved into the kill ring if that is of help at all.

I've run through this flow once and it worked, but it doesn't rule out there are other bugs.

> I fully understand that Wasabi is still in early development.

At the moment, it's more of a best effort. I had high hopes for this package, but there is relatively little interest. It took quite a bit of work to get to where it's at and would probably take additional non-trivial work to iron out all the wrinkles. Just a heads-up that I may or may not be able to fix all the bugs.

**Comment by xenodium** [comment]
ps. If you run into issues during setup, sometimes invoking `m-x wasabi-reload` gets things out of trouble.

**Comment by xenodium** [comment]
Forgot to mention... you'll need to update the repo to have the latest source and re-evaluate, at the very least wasabi.el. Lemme know if you run into issues.

**Comment by jessy836** [comment]
Hi, sorry for the slow response!

I finally got around to testing this. The `wasabi-pair-phone-number` command requires the QR code to be displayed first, but I never reach that point — `M-x wasabi` immediately shows "Something is not right".

Looking at my `*Wasabi-Log*` buffer, I see repeated errors like:

```
[2026-04-01 18:36:20] Something is wrong ((code . -32603) (message . {"level":"warn","error":"open .env: no such file or directory"...}))
```

This looks similar to #12. I think I may be blocked by that issue before I can even test the phone number pairing.

**Comment by jessy836** [comment]
I retryed today w after pulled the last commit, and iI was able to use per phone link, thanks
